tests/waku_filter_v2/test_waku_client.nim

```diff
@@ -9,720 +9,941 @@ import
+      clientSwitch = newStandardSwitch()
+      wakuFilter = await newTestWakuFilter(serverSwitch)
+      wakuFilterClient = await newTestWakuFilterClient(clientSwitch)
+
+      await allFutures(serverSwitch.start(), clientSwitch.start())
+      wakuFilterClient.registerPushHandler(messagePushHandler)
+      serverRemotePeerInfo = serverSwitch.peerInfo.toRemotePeerInfo()
+      clientPeerId = clientSwitch.peerInfo.toRemotePeerInfo().peerId
+      
+    asyncTeardown:
+      await allFutures(wakuFilter.stop(), wakuFilterClient.stop(), serverSwitch.stop(), clientSwitch.stop())
+
+    suite "Subscriber Ping":
+      asyncTest "Active Subscription Identification":
+        # Given
+        let 
+          subscribeResponse = await wakuFilterClient.subscribe(
+            serverRemotePeerInfo, pubsubTopic, contentTopicSeq
+          )
+        require:
+          subscribeResponse.isOk()
+          wakuFilter.subscriptions.hasKey(clientPeerId)
+
+        # When
+        let subscribedPingResponse = await wakuFilterClient.ping(serverRemotePeerInfo)
+
+        # Then
+        check:
+          subscribedPingResponse.isOk()
+          wakuFilter.subscriptions.hasKey(clientPeerId)
+
+      asyncTest "No Active Subscription Identification":
+        # When
+        let unsubscribedPingResponse = await wakuFilterClient.ping(serverRemotePeerInfo)
+
+        # Then
+        check:
+          unsubscribedPingResponse.isErr() # Not subscribed
+          unsubscribedPingResponse.error().kind == FilterSubscribeErrorKind.NOT_FOUND
+
+      asyncTest "After Unsubscription":
+        # Given
+        let 
+          subscribeResponse = await wakuFilterClient.subscribe(
+            serverRemotePeerInfo, pubsubTopic, contentTopicSeq
+          )
+
+        require:
+          subscribeResponse.isOk()
+          wakuFilter.subscriptions.hasKey(clientPeerId)
+
+        # When
+        let unsubscribeResponse = await wakuFilterClient.unsubscribe(
           serverRemotePeerInfo, pubsubTopic, contentTopicSeq
         )
-      assert subscribeResponse.isOk(), $subscribeResponse.error
-      require:
-        wakuFilter.subscriptions.hasKey(clientPeerId)
-
-      # When
-      let subscribedPingResponse = await wakuFilterClient.ping(serverRemotePeerInfo)
-
-      # Then
-      assert subscribedPingResponse.isOk(), $subscribedPingResponse.error
-      check:
-        wakuFilter.subscriptions.hasKey(clientPeerId)
-
-    asyncTest "No Active Subscription Identification":
-      # When
-      let unsubscribedPingResponse = await wakuFilterClient.ping(serverRemotePeerInfo)
-
-      # Then
-      check:
-        unsubscribedPingResponse.isErr() # Not subscribed
-        unsubscribedPingResponse.error().kind == FilterSubscribeErrorKind.NOT_FOUND
-
-    asyncTest "After Unsubscription":
-      # Given
-      let 
-        subscribeResponse = await wakuFilterClient.subscribe(
+        assert subscribeResponse.isOk(), $subscribeResponse.error
```

Comment:
Isn't it be unsubscribeResponse here?

Reply:
Indeed it is. Fixed, along the other typos you mentioned.

The `assert` is a suggestion from @Ivansete-status. What this does is, if the check fails it prints the value on the right, namely `$subscribeResponse.error`, which is useful for debugging.